ceph_iscsi_config/lun.py

```diff
@@ -1060,6 +1063,29 @@ def get_owner(gateways, portals):
         return sorted(portals.keys(),
                       key=lambda x: (gateways[x]['active_luns']))[0]
 
+    @staticmethod
+    def _backstore_object_name_exists(disks_config, backstore_object_name_exists):
+        return len([disk for _, disk in disks_config.items()
+                    if disk['backstore_object_name'] == backstore_object_name_exists]) > 0
+
+    @staticmethod
+    def get_backstore_object_name(pool, image, disks_config):
+        """
+        Determine the backstore storage object name based on the pool name,
+        image name, and existing storage object names to avoid conflicts
+        :param pool: pool name
+        :param image: image name
+        :param disks_config: disks configuration from `gateway.conf`
+        :return: the backstore storage object name to be used
+        """
+        base_name = '{}.{}'.format(pool, image)
+        candidate = base_name
+        counter = 0
+        while LUN._backstore_object_name_exists(disks_config, candidate):
```

Comment:
Hey, what was this for? I'm a little confused because I think we have multiple ways to create multiple disks but I do not think we hit this code path. We seem to have:

1. gwcli disk creation has a "count" argument. In this case create_disk seems to loop and allocate a index value to add to the image name. For this, only gwcli sees count and it is not passed to #2.

2. The REST call "/api/disk/<pool>/<image>" has a "count" argument. disk() seems to have a loop where it also allocates a index value to add to the image name, so we would not hit this code path that way.

3. In rbd-target-api:disk() it seems that when it calls valid_disk there is a check for if there is an existing image name in the pool, so I do not think we will loop here since we would fail in the valid_disk call already.

Maybe did you mean to merge all that stuff too?

Reply:
@mikechristie The purpose of this code is just to avoid storage object names conflicts, it's not related to multiple disks creation.

I've improved the documentation of this method to include an example that may be useful.

In summary, the storage object name will use a `.` as a separator for `pool name` and `image name`, which may cause conflicts in scenarios like this one:

  - Add disk `a.b/c` will create the storage object `a.b.c`
  - Add disk `a/b.c` will create the storage object `a.b.c.1`

Without name conflict resolution we will have an error in the second disk creation, so I think there's nothing to change in this code.

Reply:
Makes sense. LGTM.